src/H5FD.c

```diff
@@ -929,10 +929,9 @@ H5FD_cmp(const H5FD_t *f1, const H5FD_t *f2)
 {
     int ret_value = -1; /* Return value */
 
-    FUNC_ENTER_NOAPI_NOERR; /* return value is arbitrary */
+    FUNC_ENTER_NOAPI_NOERR /* return value is arbitrary */
 
-    if ((!f1 || !f1->cls) && (!f2 || !f2->cls))
-        HGOTO_DONE(0)
+        if ((!f1 || !f1->cls) && (!f2 || !f2->cls)) HGOTO_DONE(0)
```

Comment:
I'd like to find something that makes both clang and clang-format happy. This is an example where the formatting is worse when an otherwise useless semicolon is removed.

Reply:
Search current `develop` branch. This was the only `FUNC_ENTER_NOAPI_NOERR` *with* a semi colon. The other 100 or so *already* don't have a semi colon.  Given that this macro is:

```C
#define FUNC_ENTER_NOAPI_NOERR                         \
    {                                                  \
        FUNC_ENTER_COMMON_NOERR(!H5_IS_API(__func__)); \
        H5_PUSH_FUNC                                   \
        {
```

clang-format's choice isn't even so weird as this macro opens two new scopes with `{`.

Reply:
I'm sure we figure something out that will make clang-format not due weird stuff

Reply:
Commit added. You already had `StatementMacros` actually, I added items and alphabetized the list.